mambaSharedFramework/Pantos-Generic HLS Playlist Parsing/Pantos-Generic Tag Validators/EXT_X_DEFINEPlaylistValidator.swift

```diff
@@ -0,0 +1,113 @@
+//
+//  EXT_X_DEFINEPlaylistValidator.swift
+//  mamba
+//
+//  Created by Robert Galluccio on 11/30/24.
+//  Copyright © 2024 Comcast Corporation.
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License. All rights reserved.
+//
+
+class EXT_X_DEFINEPlaylistValidator: HLSPlaylistValidator {
+    static func validate(hlsPlaylist: any HLSPlaylistInterface) -> [HLSValidationIssue]? {
+        let tagValidator = EXT_X_DEFINETagValidator()
+        var validationIssues = [HLSValidationIssue]()
+        var variableNames = Set<String>()
+
+        for tag in hlsPlaylist.tags where tag.tagDescriptor == PantosTag.EXT_X_DEFINE {
+            // An EXT-X-DEFINE tag MUST contain either a NAME, an IMPORT, or a QUERYPARAM attribute, but only one of the
+            // three. Otherwise, the client MUST fail to parse the Playlist.
+            let fatalTagIssues = (tagValidator.validate(tag: tag) ?? []).filter {
+                $0.description == IssueDescription.EXT_X_DEFINENoNameNorImportNorQueryparam.rawValue ||
+                $0.description == IssueDescription.EXT_X_DEFINEMoreThanOneOfNameImportOrQueryParam.rawValue
+            }
+            validationIssues.append(contentsOf: fatalTagIssues)
+
+            var variableName: String?
+            switch DefineVariableName(tag: tag) {
+            case let .name(value):
+                variableName = value
+            case let .import(value):
+                variableName = value
+                // EXT-X-DEFINE tags containing the IMPORT attribute MUST NOT occur in Multivariant Playlists; they are
+                // only allowed in Media Playlists.
+                switch hlsPlaylist.type {
+                case .master:
```

Comment:
minor comment but maybe you can do an `if case` here for `.master` since the other cases aren't actionable

Reply:
No need to change. My style is just different 😉